test/y2security/lsm/config_test.rb

```diff
@@ -0,0 +1,251 @@
+    end
+
+    context "when no supported LSM is active" do
+      let(:active_modules) { "lockdown,capabilities,tomoyo" }
+
+      it "returns an empty array" do
+        expect(subject.active).to eq([])
+      end
+    end
+  end
+
+  describe ".from_system" do
+    let(:active_modules) { "lockdown,capabilities,apparmor" }
+
+    before do
+      allow(Yast::SCR).to receive(:Read)
+        .with(Yast.path(".target.string"), "/sys/kernel/security/lsm")
+        .and_return(active_modules)
+    end
+
+    it "returns the first supported and active LSM" do
+      expect(subject.from_system.id).to eq(:apparmor)
+    end
+  end
+
+  describe ".supported" do
+    it "returns an array with an instance of all the supported LSM" do
+      supported = subject.supported
+      expect(supported.map(&:id).sort).to eql([:apparmor, :none, :selinux])
+    end
+  end
+
+  describe ".reset" do
+    it "resets the memoized object state" do
+      supported = subject.supported
+      expect(supported).to eql(subject.supported)
+      subject.reset
+      expect(supported).to_not eql(subject.supported)
+    end
+  end
+
+  describe "#selectable" do
+    let(:selinux_selectable) { false }
+
+    it "returns an array with an instance of all the supported LSM and selectable LSM" do
+      selectable = subject.selectable
+      expect(selectable.map(&:id).sort).to eql([:apparmor, :none])
+    end
+  end
+
+  describe "#select" do
+    it "selects the LSM for the given id" do
+      expect { subject.select(:apparmor) }
+        .to change { subject.selected&.id }.from(nil).to(:apparmor)
+    end
+
+    context "when no LSM with the given id exists" do
+      it "selects nothing" do
+        expect { subject.select(:noexist) }.to_not(change { subject.selected&.id })
+      end
+    end
+
+    context "when the LSM is not selectable according to the profile" do
+      let(:selinux_selectable) { false }
+
+      it "selects nothing" do
+        expect { subject.select(:selinux) }.to_not(change { subject.selected&.id })
+      end
+    end
+  end
+
+  describe "#read" do
+    let(:normal) { true }
+    let(:from_system) { subject.supported.find { |m| m.id == :selinux } }
+
+    before do
+      allow(Yast::Stage).to receive(:normal).and_return(normal)
+      allow(subject).to receive(:from_system).and_return(from_system)
+      allow(from_system).to receive(:read) if from_system
+    end
+
+    context "when called in running system" do
+      it "selects the active module" do
+        expect { subject.read }.to change { subject.selected&.id }.from(nil).to(:selinux)
+      end
+
+      it "reads the selected LSM config" do
+        expect(from_system).to receive(:read)
+        subject.read
+      end
+
+      context "and no module is selected" do
+        let(:from_system) { nil }
+
+        it "returns false" do
+          expect(subject.read).to eql(false)
+        end
+      end
+
+      context "when called not in the running system" do
```

Comment:
Oops, this context is wrongly nested under "when called in running system" Additionally, I'd say "when not called in a running system"

Reply:
Ouch, true.